### DISCUSSION TITLE
LNP/BP standard for Bech32 common encodings

### DISCUSSION THREAD

**Discussion by dr-orlovsky** [LNP-BP/LNPBPs#85] [Proposals]
Bech32 found itself usable for visual representation of binary data and simplicity of their coping. For instance, usecases in LNP/BP software are:
- better debugging: visually bech32 shorter than hex + can be compared between logs since it's easy to find checksum and just to compare it
- better for copying data pieces between software (CLI, mobile etc) with single click

We already use it in representing RGB data ids, debug logging etc. Probably we should structure it as standards for the following cases:
1. Raw data type (`data1...`) encoding arbitrary binary data
2. Identifiers representing tagged hashes (Taproot-style) `id1...`
3. Compressed data (with ability to specify compressor version): `z1..`

Current beta reference implementation: https://github.com/LNP-BP/rust-lnpbp/blob/master/src/bech32.rs

**Comment by rajarshimaitra** [comment]
+1 on the idea. We would need a short representation of different data types and bech32 seems like a well-tested choice till now. It also seems the Bitcoin ecosystem will converge on this standard so it makes sense to use them from start.

Liked the idea of having different prefix to identify data types easily. 

Also, the implementation seems good. Would be easy to implement it by `Strategy` for all data types.

**Comment by dr-orlovsky** [comment]
Bech32 standard for LNPBP is rejected. Rationale: https://github.com/LNP-BP/rust-lnpbp/issues/226

Also related: https://github.com/LNP-BP/LNPBPs/discussions/134
